tests/test_project.py

```diff
@@ -1230,7 +1232,7 @@ def test_script(self):
         project = self.mock_project()
         # For run mode single operation groups
         for job in project:
-            job_ops = project._get_submission_operations((job,), dict())
+            job_ops = project._get_submission_operations([(job,)], dict())
```

Comment:
Can you please help me understand what you mean by this comment? Do you mean that we should avoid using private methods in our tests?

Reply:
I agree with your POV. But in my opinion (Please correct me if I'm wrong), I think we're on a road where we are really deprecating a lot of public methods and I believe in the near future, not many methods will remain in the public namespace. And if that will be the case, I think testing of private methods might become a necessity.

Reply:
I agree that is possible. However, to your point about testing, if we do end up needing to test more internal methods, we would want to write more of them as proper unit tests, i.e. tests that verify only one thing at a time. Take this test, for instance: it's simultaneously verifying 3 different internal functions, whereas we could verify both `_get_submission_operations` and `create_submission_job_operations` independently without calling `_script`. A big part of why this is happening is that our internal abstractions are not as well-defined as they need to be. I suspect that in the longer run once we have refactored the internals we will want to start exposing more components in the public namespace again; right now it's just necessary to avoid because it's not clear that we're exposing the _right_ intermediates.

In any case, this comment was not meant to be actionable in any way within this PR. For now, I think we live with this state of affairs until we can start tackling some of these refactorings.